sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -171,58 +178,76 @@ export class EventHubClient {
    */
   constructor(
     host: string,
-    entityPath: string,
+    eventHubPath: string,
     credentials: ApplicationTokenCredentials | UserTokenCredentials | DeviceTokenCredentials | MSITokenCredentials,
     options?: ClientOptions
   );
   constructor(
-    host: string,
-    entityPath: string,
-    tokenProviderOrCredentials:
+    hostOrConnectionString: string,
+    eventHubPathOrOptions?: string | ClientOptions,
+    tokenProviderOrCredentials?:
       | TokenProvider
       | ApplicationTokenCredentials
       | UserTokenCredentials
       | DeviceTokenCredentials
       | MSITokenCredentials,
     options?: ClientOptions
   ) {
-    if (!host) {
-      throw new Error(
-        "Please provide a fully qualified domain name of your Event Hub instance for the 'host' parameter."
-      );
-    }
+    let connectionString;
+    let sharedAccessKeyName = "defaultKeyName";
+    let sharedAccessKey = "defaultKeyValue";
+    let eventHubPath: string | undefined;
+    let tokenProvider = {} as TokenProvider;
```

Comment:
`sharedAccessKey`, `sharedAccessKeyName` and `tokenProvider` is used only inside the `if` block. Let's move them inside that block

Reply:
Only sharedAccessKey, sharedAccessKeyName is used inside if block. I'll move them inside the block.

Reply:
What is the reason for initializing `tokenProvider` to `{}`